inc/Engine/Admin/Settings/Page.php

```diff
@@ -75,25 +76,34 @@ class Page {
 	 */
 	private $optimize;
 
+	/**
+	 * User client instance.
+	 *
+	 * @var UserClient
+	 */
+	private $user_client;
+
 	/**
 	 * Creates an instance of the Page object.
 	 *
 	 * @since 3.0
 	 *
-	 * @param array            $args     Array of required arguments to add the admin page.
-	 * @param Settings         $settings Instance of Settings class.
-	 * @param Render_Interface $render   Implementation of Render interface.
-	 * @param Beacon           $beacon   Beacon instance.
-	 * @param Optimization     $optimize Database optimization instance.
+	 * @param array            $args        Array of required arguments to add the admin page.
+	 * @param Settings         $settings    Instance of Settings class.
+	 * @param Render_Interface $render      Implementation of Render interface.
+	 * @param Beacon           $beacon      Beacon instance.
+	 * @param Optimization     $optimize    Database optimization instance.
+	 * @param UserClient       $user_client User client instance;
 	 */
-	public function __construct( $args, Settings $settings, Render_Interface $render, Beacon $beacon, Optimization $optimize ) {
-		$this->slug       = $args['slug'];
-		$this->title      = $args['title'];
-		$this->capability = $args['capability'];
-		$this->settings   = $settings;
-		$this->render     = $render;
-		$this->beacon     = $beacon;
-		$this->optimize   = $optimize;
+	public function __construct( $args, Settings $settings, Render_Interface $render, Beacon $beacon, Optimization $optimize, UserClient $user_client ) {
+		$this->slug        = $args['slug'];
+		$this->title       = $args['title'];
+		$this->capability  = $args['capability'];
```

Comment:
We shouldn't assume that the given array of args contains these keys _when_ the class is available publicly for anyone to use. To help ensure we get what is expected:

1. Type hint to `array` for the `$args`
2. Merge the args with a default structure or use `isset` to valid or bail out if the structure isn't right